content/spin/variables.md

```diff
@@ -0,0 +1,205 @@
+
+In order to get a variable, it must be declared in an application's manifest (`spin.toml`). For example, say an application needs to access a secret. A variable named `password` could be added to the manifest and set as required with `required = true` since there is no reasonable default value for a secret. To do this, add a top-level `[variables]` section to the application manifest (`spin.toml`) and declare the variable within it.
+
+<!-- @selectiveCpy -->
+```toml
+# Add this above the [component] section
+[variables]
+password = { required = true }
+```
+
+To surface the variable to a component, add a `[component.config]` section in the component and specify the variable within it. Instead of statically assigning the value of the config variable, dynamically reference the variable with [mustache](https://mustache.github.io/)-inspired string templates. Only components that explicitly use the variables in their configuration section will get access to them. This enables only exposing variables and secrets to the desired components of an application.
+
+```toml
+# Add this below the [component.build] section
+[component.config]
+password = "\{{ password }}"
+```
+
+The resulting application manifest should look similar to the following, with the `[component.build]` section varying depending on the language used to build the `password_checker` component:
+
+<!-- @selectiveCpy -->
+```toml
+spin_manifest_version = "1"
+description = "A Spin app with a dynamically updatable variable"
+name = "password_checker"
+trigger = { type = "http", base = "/" }
+version = "0.1.0"
+
+[variables]
+password = { required = true }
+
+[[component]]
+id = "password_checker"
+source = "app.wasm"
+[component.trigger]
+route = "/..."
+[component.build]
+command = "spin py2wasm app -o app.wasm"
+[component.config]
+password = "\{{ password }}"
+```
+
+To illustrate the config API, each of the following examples receives a password via the HTTP request body, compares it to an expected password, and returns a JSON response indicating whether the submitted password matched or not. To use the [environment variable provider](/spin/dynamic-configuration.md#environment-variable-provider) to set the variable values locally, set the `password` variable's value in an environment variable prefixed with `SPIN_CONFIG_`. The provider gets the variable values from the `spin` process's environment:
+
+<!-- @selectiveCpy -->
+```bash
+$ SPIN_CONFIG_PASSWORD="123" spin build --up
+
+Serving http://127.0.0.1:3000
+Available Routes:
+  password-checker: http://127.0.0.1:3000 (wildcard)
+```
+
+Send a request to the application with the correct password in the body to authenticate successfully.
+
+<!-- @selectiveCpy -->
+```bash
+$ curl -d "123" http://127.0.0.1:3000
+{"authentication": "accepted"}
+```
+
+The exact details of calling the config SDK from a Spin application depends on the language:
```

Comment:
I do think this is a bit far from the initial "this is the config interface".  Let's break this page into sections, how to use config from an application, and how to author the config stuff in the manifest.  (I dunno which way round they should go though!)

Reply:
I created a new "Adding Variables to Your Applications" section that just focuses on the manifest. Let me know what you think

Reply:
Looks good! Thanks!